ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -153,6 +165,13 @@ a database ( directory )
           partitionsPostProcessing(iterator, scope, loadTaskTracker, tableTracker,
               partitionsTracker);
           partitionsTracker.debugLog("partitions");
+
+          if (!loadTaskTracker.hasReplicationState()) {
+            // Add ReplStateLogTask only if no pending table load tasks left for next cycle
+            ImportTableDesc tableDesc = loadPartitions.tableDesc();
+            loadTaskTracker.updateTaskCount(createTableReplLogTask(partitionsTracker.tasks(),
```

Comment:
addTask will add the tasks to the rootTasks which maintains 2 copy for ReplLogTask. one in the root and the other as child of table/partition/function tasks list. It will break the dependency. So, keeping it as it is.